app/src/main/java/cm/aptoide/pt/promotions/PromotionsManager.java

```diff
@@ -62,10 +63,18 @@ public PromotionsManager(PromotionViewAppMapper promotionViewAppMapper,
     return promotionsService.getPromotionApps(promotionId);
   }
 
-  public Observable<PromotionsModel> getPromotionsModel(String promotionsId) {
-    return getPromotionApps(promotionsId).toObservable()
-        .map(
-            appsList -> new PromotionsModel(appsList, getTotalAppc(appsList), isWalletInstalled()));
+  public Single<PromotionsModel> getPromotionsModel(String promotionsType) {
+    return promotionsService.getPromotions(promotionsType)
+        .flatMap(promotions -> {
+          if (promotions.isEmpty()) {
+            return Single.just(new ArrayList<PromotionApp>())
+                .map(__ -> PromotionsModel.ofError());
+          }
+          PromotionMeta meta = promotions.get(0);
```

Comment:
Why do you get the promotion 0 here ? weren't we supposed to send a model with all the promotions ?

Reply:
Here we are fetching the first promotion from that promotion type. If there are 2 seasonal promotions at the same time, we will show the most recent one.